iocore/net/SSLNetVConnection.cc

```diff
@@ -512,13 +512,24 @@ SSLNetVConnection::net_read_io(NetHandler *nh, EThread *lthread)
         }
       }
       // move over to the socket if we haven't already
-      if (this->handShakeReader) {
-        ink_release_assert(BIO_eof(SSL_get_rbio(this->ssl)) && !handShakeReader->is_read_avail_more_than(0));
-        // Done with the buffer after the first exchange, convert over to the socket buffer
-        BIO *rbio = BIO_new_fd(this->get_socket(), BIO_NOCLOSE);
-        BIO_set_mem_eof_return(rbio, -1);
-        SSL_set0_rbio(this->ssl, rbio);
-        free_handshake_buffers();
+      if (this->handShakeBuffer) {
+        if (BIO_eof(SSL_get_rbio(this->ssl)) && !handShakeReader->is_read_avail_more_than(0)) {
+          // Done with the buffer after the first exchange, convert over to the socket buffer
+          BIO *rbio = BIO_new_fd(this->get_socket(), BIO_NOCLOSE);
+          BIO_set_mem_eof_return(rbio, -1);
+          SSL_set0_rbio(this->ssl, rbio);
+          free_handshake_buffers();
+        } else if (handShakeReader->is_read_avail_more_than(0)) {
+          this->handShakeReader->consume(this->handShakeBioStored);
```

Comment:
Is this block same logic in other 2 places (line 447 and line 1078) in this class?
If so, it'd be great if we can avoid copy & past.